shared/models/topics/base.py

```diff
@@ -80,6 +81,7 @@ class CredentialExchange(BaseModel):
     ] = None
     # Thread id can be None in connectionless exchanges
     thread_id: Optional[str] = None
+    type: Optional[str] = None
```

Comment:
Maybe we should make this required (`type: str`) and default its value to indy. Because I presume from the changes we're making that the type must be defined, and defaulting to indy makes it backward compatible